package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,419 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+# This module written by Hugo MacDermott-Opeskin 2020
+
+r"""
+Mean Squared Displacement --- :mod:`MDAnalysis.analysis.msd`
+==============================================================
+
+:Authors: Hugo MacDermott-Opeskin
+:Year: 2020
+:Copyright: GNU Public License v2
+
+This module implements the calculation of Mean Squared Displacements (MSDs) 
+by the Einstein relation. MSDs can be used to characterize the speed at 
+which particles move and has its roots in the study of Brownian motion. 
+For a full explanation of the theory behind MSDs and the subsequent 
+calculation of self-diffusivities the reader is directed to [Maginn2019]_. 
+MSDs can be computed from the following expression, known as the 
+_Einstein_ _formula_:
+
+.. math::
+
+   MSD(r_{d}) = \bigg{\langle} \frac{1}{N} \sum_{i=1}^{N} |r_{d} 
+    - r_{d}(t_0)|^2 \bigg{\rangle}_{t_{0}}
+
+where :math:`N` is the number of equivalent particles the MSD is calculated 
+over, :math:`r` are their coordinates and :math:`d` the desired dimensionality 
+of the MSD. Note that while the definition of the MSD is universal, there are 
+many practical considerations to computing the MSD that vary between 
+implementations. In this module, we compute a "windowed" MSD, where the MSD 
+is averaged over all possible lag-times :math:`\tau \le \tau_{max}`, 
+where :math:`\tau_{max}` is the length of the trajectory, thereby maximizing 
+the number of samples.
+
+The computation of the MSD in this way can be computationally intensive due to 
+its :math:`N^2` scaling with respect to :math:`\tau_{max}`. An algorithm to 
+compute the MSD with :math:`N log(N)` scaling based on a Fast Fourier 
+Transform is known and can be accessed by setting ``fft=True`` [Calandri2011]_ 
+[Buyl2018]_.
```

Comment:
```suggestion
The FFT-based approach requires that the [tidynamics](https://github.com/pdebuyl-lab/tidynamics) package is installed; otherwise the code will raise an :exc:`ImportError`.
```

Reply:
Oops, sorry, this isn't markdown but reST... but you get the idea.